src/leap/mail/imap/mailbox.py

```diff
@@ -345,9 +244,10 @@ def getUID(self, message):
 
         :rtype: int
         """
-        msg = self.messages.get_msg_by_uid(message)
-        if msg is not None:
-            return msg.getUID()
+        # TODO --- return the uid if it has it!!!
+        d = self.collection.get_msg_by_uid(message)
+        d.addCallback(lambda m: m.getUID())
+        return d
```

Comment:
I'm confused with this function.  As the docstring states `message` is _the message uid_, I guess this is not the same than the one returned by the function. Reading the TODO I think I understand that this function is not returning the right uid, but is returning the restul of `m.getUID()`. I'm starting to think that there is three different things named uid. Can we come up with better names for the things? or at least improve the docstring?